sql/log.cc

```diff
@@ -2086,6 +2086,16 @@ static int binlog_rollback(handlerton *hton, THD *thd, bool all)
     /*
       we're here because cache_log was flushed in MYSQL_BIN_LOG::log_xid()
     */
+
+    if (cache_mngr->trx_cache.has_incident())
```

Comment:
The check should be on the level of the stmt one of l.2074 and with the same actions.
I suggest to augment that check and refine a bit the logics flow into  
```
if (cache_mngr->stmt_cache.has_incident() || cache_mngr->trx_cache.has_incident())
{ ... /* the incident branch: reset both caches regardless of who is "incidental" */ ... }
else 
{
/* do-it the rest of that is incident-free */
if (!cache_mngr->stmt_cache.empty())
...
}
```

Reply:
Actually let me correct myself to refer to [Sujatha's way](https://github.com/MariaDB/server/commit/ac6ea253a0c4b08e9d121569c992f9d986f85c71#r78996647) as the right one.